crates/sui/src/client_commands.rs

```diff
@@ -1536,6 +1504,17 @@ impl SuiClientCommands {
 
                 SuiClientCommandResult::VerifySource
             }
+            SuiClientCommands::PTB(ptb) => {
+                let args = ptb.clone().args;
+                if args.contains(&"--help".to_string()) {
+                    ptb_description().print_long_help().unwrap();
+                } else if args.contains(&"-h".to_string()) || args.is_empty() {
+                    ptb_description().print_help().unwrap();
```

Comment:
I added a TODO here. Not sure I understand what the issue is.

Reply:
At the moment, I think this is okay (no bugs here), but the pattern of checking for commands before tokenisation is generally brittle, and can run into problems when the characters you're searching for (e.g.  `-h` or `--help`) is allowed to show up inside other tokens -- the tokeniser's job is precisely to chop a sequence of characters up into tokens to deal with this complexity, and you don't get to take advantage of it if you look for the command using a raw string.

An example of a thing that may cause this to break which would be hard to detect is if we decided that it's okay to spread a single string literal across multiple shell tokens. Then I could do something like:

```
sui client ptb --assign x '"I am so' '--help' 'ful"'
```

Or maybe the less obviously wrong:

```
sui client ptb --assign x \"I am so --help ful\"
```

And it would incorrectly detect `--help` as a flag when it's part of a string.

Reply:
Excellent, let's fix it after this PR gets merged. Happy to take a stab at it.